### ISSUE TITLE
Thresholds

### ISSUE THREAD

**Issue by albertma-evotec** [jamesgleave/Deep-Docking-NonAutomated#11]
HI I am trying to understand the numbers it generated better.
In the best_models folder generated at phase 3,
there is a file named thresholds.txt
![image](https://user-images.githubusercontent.com/54853041/123269821-4126bb00-d4f7-11eb-89c7-f2c09346aa1f.png)
I understand the first number is the number of the best model and the last number is the cutoff value for dividing the validation set into hit and non-hit but what about the second number?

Thanks

**Comment by fgentile89** [comment]
Hi,
the second value us the classification threshold that is used. It is chosen in the validation set to guarantee 90% of recall (or any other recall value you specified). So, compounds with probability above that value are classified as prospective hits and added to morgan_1024_predictions, the others are classified as non-hits and discarded

Francesco

**Comment by albertma-evotec** [comment]
Sorry if I misunderstand this but it sounds like the third number (-68.02) is what you described above refering to the code. 

![image](https://user-images.githubusercontent.com/54853041/123298723-b2bf3300-d510-11eb-9901-423512c3f171.png)

I am a bit confused with the threshold vs cutoff in the code.

**Comment by fgentile89** [comment]
Hi,
yes that may be a little bit confusing, so let me explain it in details:
- in the validation set, the code ranks the molecules by their docking scores, and label as positive a certain percentage at the top (for the first iteration, we usually set it to 1% using the arguments of simple_job_models.py)
- Then it checks the score associated with the last of this 'positive' set. This is what cf_to_use is
- Then, the model is trained by using this score cutoff to label positive and negative in the training set, and the resulting model is used to assign a virtual hit-likeness probability to each molecule in the validation set 
- Now, the validation set is ranked by these probabilities. Since it is known which compounds are true positive, a probability threshold is selected, in order for 90% of the true positives to be above it. This is the main_threshold
- Then, inference is run on the whole dataset and compounds with virtual hit-likeness above the threshold are retained, others are discarded (of course before the models are evaluated on the test set for generalizability)

By the way, we are preparing a technical paper which describes the protocol in details, as we know that the code is not easy to read and understand.

Hope this helps, cheers

Francesco

**Comment by albertma-evotec** [comment]
Really great explanation and very helpful indeed. I am looking forward to the technical paper. Thanks for the great work!

**Comment by fgentile89** [comment]
Thank to you for using our code!

Francesco
